CONTRIBUTING.md

```diff
@@ -0,0 +1,75 @@
+- So that this software can be liberally used with few restrictions
+
+After ensuring the license options are compatible with the aims of the contribution, then please submit your PR for review and we will review as soon as possible. My only ask is that you do not do this for free, unless it's something that is a passion or learning project for you. Please, find a way to be paid for your work. You are worth it.
+
+## Understanding the design
+
+Please read the [Architecture](ARCHITECTURE.md) to understand the general design and layout of the Trust-DNS projects.
+
+## Submitting PRs
+
+Before submitting a PR it would be good to discuss the change in an issue so as to avoid wasted work, also feel free to reach out on the Discord channel listed on the front page of the GitHub project. Please, consider keep PRs focused on one issue at a time. While issues are not required for a PR to be accepted they are encouraged, especially for anything that would change behavior, change an API, or be a medium to large change.
+
+If changes will span multiple crates, consider breaking up the PR into multiple commits such that a reviewer can follow the change. This is no necessary, but can make it easier for a reviewer to follow the changes.
+
+### Test policy
+
+All PRs *must* be passing all tests. Ideally any PR submitted should have more than 85% code coverage, but this is not mandated. When tests are failing, especially on previous branches this is often due to checked in testing keys for the DoH and DoT tests. See **Updating Security Related Tests**.
+
+## Releases
+
+Trust-DNS tries to follow semver versioning semantics. Major versions will not break APIs in a current major revision. If changes are being made to the current `main` branch, double check the current status of the Major release. Until `1.x.x`, all `0.x.x` minor releases are treated as major releases with breaking changes allowed. Releases are performed on an ad-hoc/on-demand basis.
+
+*Maintainers*: If changes are needed to previous releases, then there should exist a `releease/x.x`. If this does not exist, then go to the previous most recent tag (release) and create a new branch at that tag `release/x.x`, for example the branch `release/0.19`:
+
+```shell
+> git fetch origin
+> git checkout v0.19
+> git branch release/0.19
+> git push --set-upstream origin release/0.19
+```
+
+Previous release can fail due to time-lapse, please see **Updating Security Related Tests**.
+
+## Performing a Release, for Maintainers
+
+Releases are somewhat automated. The github action, `publish`, watches for any tags on the project. It then attempts to perform a release of all the libraries, this does not always work, for various reasons.
+
+1. Create a new branch like `git checkout -b prepare-0.20.1`
+2. Update all Cargo.toml files to the new version, `version = 0.20.1`
+3. Update all inter-dependent crates, i.e. trust-dns-resolver to use `trust-dns-proto = 0.20.1`
```

Comment:
Add a step here to run `cargo update`? I suppose this needs some care not to bump public dependencies.

Reply:
I think cargo update only does minor updates, so we won't say bump to versions that would be incompatible in public APIs or example.